hazelcast/src/main/java/com/hazelcast/jet/impl/util/Util.java

```diff
@@ -140,6 +146,22 @@ public final class Util {
     private Util() {
     }
 
+    public static boolean checkJobIsAllowedToBeSuspended(TerminationMode terminationMode, JobConfig jobConfig) {
+        if (isJobSuspendable(jobConfig)) {
+            return true;
+        }
+
+        return !(terminationMode == SUSPEND_GRACEFUL
+                || terminationMode == SUSPEND_FORCEFUL
+                || terminationMode == RESTART_GRACEFUL
+                || terminationMode == RESTART_FORCEFUL);
```

Comment:
This is equivalent to `terminationMode.actionAfterTerminate() == CANCEL` and it reads as "the job can be suspended if it is about to be cancelled," which does not make sense. `checkJobIsAllowedToBeSuspended` should be removed, this condition (or the simpler one I gave) should be moved to `MasterJobContext.requestTermination()`.

Reply:
Good catch!

Reply:
https://github.com/hazelcast/hazelcast/pull/25939/commits/a4a330ace4f10041b8ca7c05441edb8d8e779adf

Reply:
It still reads as "the job can be suspended if it is about to be cancelled."